### PR TITLE
Node ui

### PR DESCRIPTION
**Pull request by hannessolo**
DB provider is connected to Graph component using dependency injector

### CODE REVIEW

File: src/components/LocalDBForm.test.tsx [hunk 1]
```diff
@@ -27,10 +27,46 @@ it('navigates to pathname equal to input to LocalDBForm', () => {
...
+  const errorMessage = await findByTestId('errorBox')
+
+  expect(errorMessage).not.toBe(null);
+
+});
+
+it('navigates to correct url on valid address', () => {
+  const validAddress = '/orbitdb/QmfY3udPcWUD5NREjrUV351Cia7q4DXNcfyRLJzUPL3wPD/hello'
+  const history = createMemoryHistory();
+  const {container} = render(
```
Comment (justincqz) [inline src/components/LocalDBForm.test.tsx]: Why doesn't the linter pick up this?

File: src/components/useDBFormHandler.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  return {
+    error: errorMessage,
+    address: address,
+    setAddress: setAddress,
+    submit: submit
+  }
+}
+
+export default useDBFormHandler;
```
Comment (justincqz) [inline src/components/useDBFormHandler.ts]: Missing EOF new line?

File: src/index.css [hunk 3]
```diff
@@ -11,3 +9,16 @@ code {
...
+    background-image:none;
+    background-color:transparent;
+    -webkit-box-shadow: none;
+    -moz-box-shadow: none;
+    box-shadow: none;
+}
+
+input:focus {
+  outline: none;
+}
```
Comment (justincqz) [inline src/index.css]: This too

File: src/pages/Database.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import GraphDisplay from '../components/GraphDisplay';
+import { useDependencyInjector } from '../state/dependencyInjector';
+import { D3Data } from '../utils/D3Data';
+ 
+const DatabaseView: React.FC = () => {
+  // URL parameters
+  let {hash, name}: {hash: string, name: string} = useParams();
```
Comment (justincqz) [inline src/pages/Database.tsx]: `let { hash, name }`

File: src/utils/DAGNode.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      })
+    ])
+  )
+});
+
+it('correctly limits output size', () => {
+  const limit = 1;
+  let d3data = node.toD3Data(limit);
+  expect(d3data.length).toEqual(1);
+})
```
Comment (justincqz) [inline src/utils/DAGNode.test.ts]: EOF newline

File: src/utils/DAGNode.ts [hunk 6]
```diff
@@ -1,29 +1,60 @@
...
+        id: node.hash,
+        parentIds: []
+      };
 
-    static createDAG(head: any, db: Store, depth: number = 10): DAGNode {
-        return depth > 0
-            ? new DAGNode(
-                head.hash,
-                (head.next
-                    ? head.next.map(node => this.createDAG(db.get(node), db, --depth))
```
Comment (justincqz) [inline src/utils/DAGNode.ts]: I think you and matthew literally fixed this in two different branches rip

Comment (justincqz) [reply src/utils/DAGNode.ts]: Nevermind its at the bottom didn't see it cause git can't diff properly

File: src/index.css (hunk 3 above)
Comment (justincqz) [reply src/index.css]: Add the EOF newline

File: src/components/LocalDBForm.test.tsx (hunk 1 above)
Comment (justincqz) [reply src/components/LocalDBForm.test.tsx]: `const { container } = ...`

File: src/pages/Database.tsx (hunk 4 above)
Comment (hannessolo) [reply src/pages/Database.tsx]: ?

File: src/components/LocalDBForm.test.tsx (hunk 1 above)
Comment (hannessolo) [reply src/components/LocalDBForm.test.tsx]: that was hadrian, check git blame ;)

File: src/pages/Database.tsx (hunk 4 above)
Comment (hannessolo) [reply src/pages/Database.tsx]: oh

**Pull request merged**